### PR TITLE
Nemotron export - fixing megatron_export.py 

### PR DESCRIPTION
**Pull request by borisfom**
# What does this PR do ?

Updates Apex module replacements we do inside export() with Megatron Core modules replacement.

**Collection**: [Note which collection this PR will affect]

NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage

```python
python3 examples/nlp/language_modeling/megatron_export.py model_type=gpt gpt_model_file=Nemotron-3-8B-Base-4k.nemo onnx_model=Nemotron-3-8B-Base-4k.nemo
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [X ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/nlp/language_modeling/megatron_export.py [hunk 1]
```diff
@@ -74,7 +75,7 @@ def nemo_export(cfg):
     assert nemo_in is not None, "NeMo model not provided. Please provide the path to the .nemo or .ckpt file"
 
     onnx_out = cfg.onnx_model_file
-
+    print(f"onnx_out: {onnx_out}")
```
Comment (ericharper) [inline examples/nlp/language_modeling/megatron_export.py:78]: Is this needed? Either remove or at least use logging

File: examples/nlp/language_modeling/megatron_export.py [hunk 2]
```diff
@@ -149,18 +150,28 @@
     try:
         model.to(device=cfg.export_options.device).freeze()
         model.eval()
+
+        sequence = "sequence"
+        batch = "batch"
+
+        use_dynamo = False
+        if use_dynamo:
+            sequence = Dim("sequence")
```
[the remark this answers is not in the document]
Comment (ericharper) [reply examples/nlp/language_modeling/megatron_export.py:159]: Seems like a legit issue?

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!
